### ISSUE TITLE
Runs Saving in temp memory when training in databricks for YOLOV8

### ISSUE THREAD

**Issue by sivaji123256** [ultralytics/ultralytics#21595]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi  , I am training a YOLOv8 model in Azure Databricks notebook. Ideally, when I don't specify the save_dir, the model should save in the runs folder within the workspace. However, it is currently saving to a temporary NFS or in-memory location that isn't persistent and cannot be used. Additionally, MLflow is not functioning correctly in Databricks. There appears to be a conflict between the MLflow version integrated with Databricks and the one from Ultralytics. Despite explicitly setting mlflow=True, no MLflow experiment is being created in Azure Databricks. How can I resolve these issues?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sivaji123256, thank you for reaching out and sharing these details about your Databricks training environment 🚀! This is an automated response to help get you started—an Ultralytics engineer will review your issue and assist you soon.

We recommend checking our [Docs](https://docs.ultralytics.com/) for helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions and best practices are covered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us investigate further. This is especially useful for environment-specific issues like yours. If possible, include:
- The code snippet or notebook cell you're using to train your YOLOv8 model in Databricks
- Your Databricks runtime version and Python version
- Any relevant logs or stack traces

For questions about custom training, sharing additional details such as your workspace setup, dataset structure, and training logs will help us provide more tailored support. Please also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for more support and discussion:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth topics: [Discourse](https://community.ultralytics.com/)
- Community threads: [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

To ensure compatibility and access to the latest features and bug fixes, upgrade the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) within a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience while we investigate your Databricks + MLflow integration issue!

**Comment by akx** [comment]
Please don't tag arbitrary people in your issues...

**Comment by glenn-jocher** [comment]
@sivaji123256 For the save directory issue, explicitly set a persistent path using the `project` argument when training: `model.train(data="your_data.yaml", project="/dbfs/FileStore/yolo_runs")` to ensure runs save to DBFS which persists across sessions.

For MLflow conflicts, try disabling Ultralytics' MLflow integration with `yolo settings mlflow=False`, then use Databricks' native MLflow directly in your training script with `import mlflow` and manual logging calls around your training code.

**Comment by sivaji123256** [comment]
Hi @glenn-jocher , Thanks for your response. I disabled the ultralytics ml flow to false and I’m looking to log complete results like the ml flow server experiment in Dtaabricks. Could you share any relevant code for this ? Thanks.

**Comment by glenn-jocher** [comment]
@sivaji123256 Here's how to manually log complete results using Databricks' native MLflow:

```python
import mlflow
from ultralytics import YOLO

# Start MLflow run
with mlflow.start_run():
    model = YOLO("yolo11n.pt")
    
    # Log training parameters
    mlflow.log_params({
        "epochs": 100,
        "imgsz": 640,
        "batch": 16,
        "data": "your_data.yaml"
    })
    
    # Train model
    results = model.train(
        data="your_data.yaml", 
        epochs=100,
        project="/dbfs/FileStore/yolo_runs"
    )
    
    # Log final metrics
    if hasattr(results, 'results_dict'):
        mlflow.log_metrics(results.results_dict)
    
    # Log model artifacts
    mlflow.log_artifacts(str(model.trainer.save_dir))
    mlflow.pytorch.log_model(model.model, "model")
```

This will create a complete MLflow experiment in Databricks with parameters, metrics, and model artifacts similar to the automatic integration.
